src/App.tsx

```diff
@@ -8,37 +8,95 @@ import { PostsList } from './components/PostsList';
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isForm, setIsForm] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    service.getUsers()
+      .then(use => setUsers(use));
+  }, []);
+
+  const handleListDropDown = () => {
+    setIsDropDownList(!isDropDownList);
+  };
+
+  const handleChoseUser = (user: User) => {
+    setSelectedUser(user);
+    setIsLoading(true);
+    setIsDropDownList(false);
+    setSelectedPost(null);
+
+    service.getPosts(user.id)
+      .then(pos => setPosts(pos))
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  handleListDropDown={handleListDropDown}
+                  selectedUser={selectedUser}
+                  isDropDownList={isDropDownList}
+                  setIsDropDownList={setIsDropDownList}
+                  users={users}
+                  handleChoseUser={handleChoseUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {selectedUser && !isLoading && !posts.length && (
+                  <div
+                    className="notification is-warning"
+                    data-cy="NoPostsYet"
+                  >
+                    No posts yet
+                  </div>
+                )}
 
-                <PostsList />
+                {selectedUser && posts.length > 0 && !isError && (
```

Comment:
```suggestion
                {selectedUser && !!posts.length && !isError && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**